oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/event/OAuth2AuthorizationEvent.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.event;
+
+import org.springframework.security.access.event.AbstractAuthorizationEvent;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+
+public abstract class OAuth2AuthorizationEvent extends AbstractAuthorizationEvent {
```

Comment:
`AbstractAuthorizationEvent` are events that are fired by `FilterSecurityInterceptor` when authorizing the current request. Instead, we should extend from `AbstractAuthenticationEvent` for authentication related events.

Take a look at the hierarchy of `AbstractAuthenticationEvent` in order to determine where to extend from. The common extensions will be either `AuthenticationSuccessEvent` and `AbstractAuthenticationFailureEvent`.

At this point, I'm not sure if we'll introduce an OAuth2-specific event hiearchy, so for now let's just prefix the event classes with OAuth2.

Reply:
When a token is issued, it isn't an authentication event. As such this feels a bit forced, but I have made the change.